Tchap/Modules/Application/AppCoordinator.swift

```diff
@@ -143,6 +220,103 @@ final class AppCoordinator: AppCoordinatorType {
         NotificationCenter.default.removeObserver(self, name: NSNotification.Name.mxSessionDidCorruptData, object: nil)
     }
     
+    private func handleRegisterAfterEmailValidation(_ registerParams: [String: String]) {
+        // Check required parameters
+        guard let homeserver = registerParams["hs_url"],
+            let sessionId = registerParams["session_id"],
+            let clientSecret = registerParams["client_secret"],
+            let sid = registerParams["sid"] else {
+                NSLog("[AppCoordinator] handleRegisterAfterEmailValidation: failed, missing parameters")
+                return
+        }
+        
+        // Check whether there is already an active account
+        if self.mainSession != nil {
+            NSLog("[AppCoordinator] handleRegisterAfterEmailValidation: Prompt to logout current sessions to complete the registration")
+            AppDelegate.theDelegate().logout(withConfirmation: true) { (isLoggedOut) in
+                if isLoggedOut {
+                    self.handleRegisterAfterEmailValidation(registerParams)
+                }
+            }
+            return
+        }
+        
+        // Create a rest client
+        let restClientBuilder = RestClientBuilder()
+        restClientBuilder.build(homeserver) { (restClientBuilderResult) in
+            switch restClientBuilderResult {
+            case .success(let restClient):
+                if let identityServerURL = registerParams["is_url"] {
+                    restClient.identityServer = identityServerURL
+                }
+                
+                guard let identityServer = restClient.identityServer,
```

Comment:
Use `guard let identityServerURL = registerParams["is_url"]` and set `restClient.identityServer` below if needed

Reply:
I don't see what you mean, can you please write here your suggestion?

Reply:
Yes, I would say:
```
    guard let identityServerURL = registerParams["is_url"],
    let identityServerURL = URL(string: identityServer),
    let identityServerHost = identityServerURL.host else {
        let error = NSError(domain: MXKAuthErrorDomain, code: 0, userInfo: [NSLocalizedDescriptionKey: Bundle.mxk_localizedString(forKey: "error_common_message")])
        // FIXME: Present an error on coordinator.toPresentable()
        AppDelegate.theDelegate().showError(asAlert: error)
        return
    }
    
    restClient.identityServer = identityServerURL
```

Reply:
Ok I understood what you mean. But here I set on purpose the potential id server before getting the restClient.identityServer in order to get the default id server (which is the homeserver) when no id server is set)